drivers:platform:stm32:pwm: Add PWM Slave mode

NagaHimanshu opened the pull request:
## Pull Request Description

Added Slave modes for timers to select trigger, external1 mode and none.
Fixed previous configuration retention in timers. When timers are reinitialized i.e. pwm_remove and then pwm_init called, previous configuration is getting retained for one-pulse mode.

Also Fixes: 9264d8983c4e ("drivers/platform/stm32: stm32 pwm support.")
Also Fixes: 247be2df09cf("platform: stm32: pwm: Add onepulse option")

## PR Type
- [x] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [x] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

buha commented on projects/ad463x_fmcz/src/platform/stm32/parameters.c:
```diff
@@ -125,7 +125,7 @@ struct stm32_pwm_init_param tx_pwm_extra_init_params = {
 	.clock_divider = 2,
 	.onepulse_enable = true,
-	.trigger_enable = true,
+	.slave_mode	= STM32_PWM_SM_TRIGGER,
```
don't introduce tabs here if the others dont have tabs

NagaHimanshu commented on drivers/platform/stm32/stm32_pwm.h:
```diff
@@ -51,6 +51,7 @@ enum TimOCMode {
 };
 
 enum stm32_pwm_trigger {
+	PWM_TS_NONE,
```
Yes correct. But I thought it wouldn't be good to place NONE in end of enum.

NagaHimanshu commented on drivers/platform/stm32/stm32_pwm.c:
```diff
@@ -402,7 +425,15 @@ static int32_t stm32_init_lptimer(struct stm32_pwm_desc *desc,
 			return -EINVAL;
 		}
+		break;
+	case STM32_PWM_SM_EXTERNAL1:
```
Right now the implementation for external mode for lptimers is not supported in the driver. It may be supported by peripheral and HAL (not sure, need to explore).

NagaHimanshu commented:
V2:
- Removed the tabs around '=' in affected project files.
- Added comment 'Not supported' for EXTERNAL mode for LPTIM and also returning error code ENOTSUP.
